### DISCUSSION TITLE
Normal PW Gateway behavior?

### DISCUSSION THREAD

**Discussion by ViktorJp** [jasonacox/Powerwall-Dashboard#465] [Q&A]
Hi All... just a general question of PW Gateway 2 I've behavior I've been seeing since day 1 of interacting with it via API... hope you all don't mind, but Tesla PW support can be difficult to deal with at times (I'm sure you understand). ;)

My Gateway connectivity goes down multiple times per day, completely.  The unit is still working, ie. I'm getting telemetry from the Tesla app just fine.  It's just that the local IP of the gateway interface no longer responds at all.  You can't ping it.  You can't connect to it.  I built my own app that monitors the GW through a SSH script, and it also goes down.  The Powerwall Dashboard will show gaps in coverage, but they're typically short-lived.  Sometimes a few minutes to 10-15mins.  But then the PW bounces back and everything is normal.

I asked Tesla support, thinking that there might be something wrong with my GW, or its network interface... Their explanation is that "probably" when the WiFi signal is too weak, it flips to cellular, causing a communications disruption.  I've called their BS.  The Gateway is literally 20' away from an access point, and it's not going down. I asked them to look at interface logs to see what was causing it to go down, but who knows if they did... it's like pulling teeth sometimes with them.

Just wondering if anyone else has seen this behavior, or what you may have been able to do to mitigate this?

**Comment by jasonacox** [comment]
HI @ViktorJp , great question and you are not alone. My $0.02:

* The WiFi transceiver in the Powerwall Gateway is weak. It is buried in a near Faraday cage of wires, metal shielding, conduit, circuit boards and high voltage connectors. I suspect that is the root cause of the weak performance. An external antenna may help. My system has a Neurio WiFi CT that is used to measure solar power production on the Solar Inverter and is 1.5 foot away from the Gateway WiFi built in transceiver access point. It loses connection several times a day.
* Like you, I have an access point less than 20 feet away from my Powerwall Gateway yet when it was connected to WiFi it would drop 5-10 min. at a time, several times a day.  I wanted constant connection so I hard wired into the ethernet port. I no longer see any dropped packets, though the Neurio continues to lose connection since it still uses the Gateway WiFi.
* Having said the above, several in this community have reported zero issues with their WiFi connection. I don't know if it's indicative of different hardware in the Gateway or just the installation, placement and WiFi signal geometry that makes it better.
* It sounds like your issue is similar to mine and related to the weak WiFi support in the Gateway. However, there is another issue that you can see even hardwired. The local API of the Gateway was never meant for "production".  It is super handy and a great backup for when all internet connection and power is down. Naturally, we also love it for the additional data we can get from the system which competing vendors don't provide. Having said that, it has limited CPU power to process request. To protect that, the engineers build in a rate limiting algorithm to mitigate high CPU load conditions. If you have too many polling requests from too many sources, it wills start to deny connections (cooldown mode). The pypowerwall proxy tries to help with that by caching payloads for redundant calls and using connection pooling to limit the TCP setup/teardown load on the Gateway. There were bugs in older pypowerwall version that caused retry loops for depreciated APIs (e.g. vitals API) that would create this condition so I would recommend everyone upgrading to latest. Additionally, having too many other scripts that talk to the Powerwall directly (e.g. Home Assistant, manual cron jobs, other monitoring tools) could contribute to load on the Gateway.

If you figure out how to make the WiFi work better, please let us know. Otherwise, you may want to consider running an ethernet cable.

**Reply by ViktorJp** [reply]
That's great info @jasonacox!  I think I'm going to probably have to live with the problem.  It would be too much of a PITA to hardwire this given that it only happens occasionally throughout the day... it's not a showstopper in other words.

I did bring down the polling interval of my own PW monitoring SSH script down from every 10 seconds to every 60 seconds just to see if that made a difference... but was greeted with a non-responsive PW gateway this morning as soon as I logged in.  So that didn't seem to help.  ;)  I truly think you're right... it's got to be some kind of weak wifi signal, or perhaps by design, it fails over to mobile every so often to ensure that route still works.  

Thanks much for your help with this! :)

**Reply by jasonacox** [reply]
Happy to help! I do wish there were ways to externalize the WiFi antenna on the gateway.

One thing I should have mentioned and suspect you have already tried, make sure you lock your Powerwall to the same DHCP address in your WiFi router/access point. Some have reported that the network ARP process that will occur at the lease expire and renegotiation can be some of the problem.

**Comment by jgleigh** [comment]
Had tons of issues with my Gateway when I was on WiFi. I don't think it likes dual-band routers and it would actually lockup the Gateway computer and my whole system would stop producing. I connected it through Ethernet and it's been MUCH more stable.

**Comment by emptywee** [comment]
To add to that, yesterday in the morning the tesla gateway stopped responding to anything that is not on the same subnet. I place all IoT/Smart/etc stuff in its own subnet and isolate it from the home LAN, where all my monitoring/rpi/VM/infra is. So I am waking up on a Sunday morning to see that tesla gw is down, no telemetry in grafana for the powerwall dashboard, etc. I see it reporting metrics to tesla cloud though, and all stuff on the same subnet can ping it and access its API. 

Ended up NAT'ing it on the router, but it's disgusting to know that tesla can basically do whatever they want with something you purchased and own at any time they please. Boo at Tesla and Elon!

**Reply by jasonacox** [reply]
Yikes. Do you see anything new/interesting in http://localhost:8675/api/networks ?

**Reply by emptywee** [reply]
Nothing out of ordinary, tbh.
